### ISSUE TITLE
Ceiling fans

### ISSUE THREAD

**Issue by shorowit** [NREL/OpenStudio-BEopt#118]
- [x] Do we want to include BEopt's "Use Benchmark Energy" input in our measure? Yes, this input is included. Selecting "yes" creates the ceiling fan energy portion of MELs and zeros out the EMS electric equipment object.
- [x] What value is assigned to `unit.misc_electric_loads.rad`? This is Fraction Radiant for the ceiling fan electric equipment object. The constants.rb resource file has been updated.
- [x] Uses `Tin` sensor in the EMS program. The airflow measure also uses this sensor. How should we handle not duplicating this object, etc.? At the beginning of both the airflow and ceiling fan measures, I remove existing output variables, sensors, etc. with the "residential airflow ..." or "residential ceiling fan ..." name prefixes.

@jmaguire1 @shorowit

**Comment by jmaguire1** [comment]
As far as where EMS comes into play for ceiling fans, it looks like we only use that to turn fans on or off based on the indoor air temperature, with the ceiling fans being forced off when the living space temperature is below the minimum space setpoint minus 0.1 (so it'll stay on if the fan if the space stays at the cooling setpoint temperature). See below for the EMS code:

EnergyManagementSystem:Program,
   CeilingFanScheduleProgram_1,                !- Name
   If CeilingFan_sch_1 == 0,
       Set CeilingFanScheduleOverride_1 = 0,
   ElseIf Tin_1 <24.3888888889,
       Set CeilingFanScheduleOverride_1 = 0,
   Else,
       Set CeilingFanScheduleOverride_1 = 1,
   EndIf;

There's some more complicated schedule stuff that happens in terms of setback, but it's not anything we have to handle in EMS. If we want to keep this temperature dependent nature of the ceiling fan (rather than just having it run during the whole cooling season, maybe there's another option I'm not thinking of?) I think we need to keep this little bit of EMS.

Ceiling fans also can impact the cooling setpoint, so I think we'll always need to run the ceiling fan measure after that's been run.

**Comment by jmaguire1** [comment]
A few more notes on ceiling fans, based on an analysis of 4 climates (Albequerque, Houston, Los Angeles, and Phoenix):

- As they're set up now, ceiling fans can run during times that aren't the cooling season, they only depend on the indoor air temperature. >90% of the time that they're on is during the cooling season though.
- Depending on the climate, the ceiling fans run for 51-89% of the cooling season. So using the cooling season schedule to control the ceiling fans instead of indoor air temperature doesn't seem like a good proxy. 

We could try to relate runtime to hourly or daily ODB temperature instead if we want to get rid of the EMS we have controlling them.

**Comment by shorowit** [comment]
Thanks for investigating, @jmaguire1. Let's stick with the above CeilingFanScheduleProgram EMS program then. @joseph-robertson, I think this is ready for your implementation, let us know if you have any questions.

Also, @jmaguire1, I'm not seeing additional EMS related to complicated schedules and setbacks. Can you point me to it?

**Comment by jmaguire1** [comment]
On further investigation, there isn't any additional EMS. The way ceiling fans are controlled is that CeilingFan_sch_1 is 0 for any hours with setup, so the above EMS code will force the fans off during those hours BEFORE doing any looking at the space temperature. For hours without setup the controls are done on a temperature basis with the ceiling fan on when the space temp is at or near the cooling setpoint.

**Comment by joseph-robertson** [comment]
- [ ] What value is assigned to `unit.misc_electric_loads.rad`? This is Fraction Radiant for the ceiling fan electric equipment object.

**Comment by jmaguire1** [comment]
For the sensible/latent breakdown (from sim.py):

mel.sens = 0.93
mel.lat = 0.021

mel.conv = mel.sens / 2.5
mel.rad = mel.conv * 1.5

There's a few special cases in sim.py where we change to convective/radiative split, but we don't care about covering any of those for now

**Comment by shorowit** [comment]
Yes, we want to include the Use Benchmark Energy argument. Our default option uses it (and it's the simplest to implement as it doesn't involve the EMS or cooling setpoint offset).

**Comment by jmaguire1** [comment]
I should have posted the full code,  we use the same sensible/latent split whether CeilingFanUseBenchmarkEnergy is true or false:

        if self.Standard140_mode:
            mel.sens = 0.822
            mel.lat = 0.178
            
        elif unit.ceiling_fans.CeilingFanUseBenchmarkEnergy:
            # Use the value for MELs specified in the BA Benchmark (which includes a contribution 
            # from ceiling fans). This should only be YES when user selects the "Benchmark"
            # option for ceiling fans.

            mel.sens = 0.93
            mel.lat = 0.021

        else:
            # Modify the Sensible and Latent Fractions of the BA Benchmark to subtract the ceiling 
            # fan contributions, which only affect the variable MELs. In this case, the ceiling fan 
            # sensible and latent loads are specfied separately.
            # The modified fractions were determined using the ba_analysis spreadsheet and changing 
            # the "Fractional Units per Household" for ceiling fans from 1.84 to 0.
            # The sensible and latent fractions were calculated as ratios of the Adjusted Sensible 
            # Load and Adjusted Latent Load to the Adjusted Benchmark MEL Energy (which are all 
            # calculated in the spreadsheet).
            # The "Fixed (All Elect)" values were used for FixedMLSens and FixedMLLat.
            # mel.sens = 0.723
            # mel.lat = 0.02

            mel.sens = 0.93
            mel.lat = 0.021

It looks like this was changed years ago by Paulo and never fully removed?

**Comment by jmaguire1** [comment]
Also, in energyplus.py we set the latent fraction to 0 (which makes sense, I wouldn't expect a ceiling fan to put out any moisture) but don't change the sensible fraction at all to compensate. Probably not that big a deal.

If I were coming up with these fractions now, my first assumption would be that 100% of the power consumed by the ceiling fan becomes a convective gain to the space unless I saw any evidence to the contrary.
